File: client-app/src/app/models/post.ts
import { Profile } from "./profile";

export interface Post {
    id: string;
    title: string;
    date: Date | null;
    description: string;
    category: string;
    venue: string;
    hostUsername: string;
    isCancelled?: boolean;
    isGoing: boolean;
    isHost: boolean;
    host?: Profile;
    attendees: Profile[];
}

export class PostFormValues {
    id?: string = undefined;
    title: string = '';

}

File: Domain/Post.cs
using System;
using System.Collections.Generic;

namespace Domain
{
    public class Post
    {
        public Guid Id { get; set; }
        public string Title { get; set; }
        public DateTime Date { get; set; }
        public string Description { get; set; }
        public string Category { get; set; }
        public string Venue { get; set; }
        public bool IsCancelled { get; set; }
        public ICollection<PostAttendee> Attendees { get; set; } = new List<PostAttendee>();
        public ICollection<Comment> Comments { get; set; } = new List<Comment>();
    }
}